### PR TITLE
FE connect BE authorization/login page

### PR DESCRIPTION
**Pull request by AlexeyBezrebriy**
fix: DREAMTEM-3-Basic UI kit components## Describe your changes

- I added interceptors for auto sending access token and updating access token via refresh token.
- I changed login form to new design.
- I fixed some simple bugs in components.

## Issue ticket code (and/or) and link

- [DREAMTEAM-62](https://zenbitdreamteam.atlassian.net/browse/DREAMTEAM-62)

### **General**
image
image
image
image
image

- [x] Types for input and output params
- [x] No `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no magic numbers (describe it in constants)
- [x] compare only with constants not with strings (instead of user === ‘adminʼ, better user === roles.admin)
- [ ] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no console.log in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL should be in constant file
- [x] constants and function names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyyˮ), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _ʼ, const EACH_DAY_15h_15min = ‘15 3 _ \* \*ʼ

### Frontend

- [x] split component and logic (move logic to custom hook file)
- [ ] colors, font size, and font name should be in the theme or in the constants
- [ ] no text in the components, use i18n approach, even if you have only one language for now
- [ ] inline styles prohibited
- [x] import should be absolute. instead of ../../../components/myComponent should be components/myComponent

### CODE REVIEW

File: src/assets/styles/constants/colors.ts [hunk 1]
```diff
@@ -1,56 +1,59 @@
...
+  outline: '#79747E',
+  outlineVariant: '#CAC4D0',
+
+  inverseSurface: '#322F35',
+  inverseOnSurface: '#F5EFF7',
+  scrim: '#000000',
+  shadow: '#000000',
+
+  disabled: '#D0D5DD',
+  signInContainer: '#F8F7FA',
```
Comment (lyudad) [inline src/assets/styles/constants/colors.ts]: I already see this changes in Olexandr PR. Are you doing the same things? I mean eslint rules

Comment (AlexeyBezrebriy) [reply src/assets/styles/constants/colors.ts]: I didn’t see what he did, so we worked on the same bugs and fixed the same issues

File: src/components/Button/ArrowButton/styles.ts [hunk 2]
```diff
@@ -1,54 +1,72 @@
...
+  defaultButtonSize,
+  ButtonSize,
+} from '@/assets/styles/constants/buttonConstants';
+import { COLORS } from '@/assets/styles/constants/colors';
+import { FONTS } from '@/assets/styles/constants/fonts';
 
 type ButtonVariantType = keyof typeof buttonVariants;
 type ButtonColorType = keyof typeof buttonColors;
 
+// Constants for opacity and transition
```
Comment (lyudad) [inline src/components/Button/ArrowButton/styles.ts]: I think this comment useless, by the name of the variable its clear what they done

File: src/components/Button/CreateRouteButton/index.tsx [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
-    return (
-        <CreateRouteBtn>
-            <span style={{ color: "white", marginRight: "10px" , fontSize: "25px"}}>+</span>
-            Create route
-        </CreateRouteBtn>
-    );
+export const CreateRouteButton = (): JSX.Element => {
+  return (
+    <CreateRouteBtn>
+      <span style={{ color: 'white', marginRight: '10px', fontSize: '25px' }}>
```
Comment (lyudad) [inline src/components/Button/CreateRouteButton/index.tsx]: 1.3.2

File: src/components/Button/SignIn/styles.ts [hunk 4]
```diff
@@ -1,19 +1,44 @@
...
-    margin-top: 14px;
-    border: none; 
-    box-sizing: border-box; 
-`;
\ No newline at end of file
+  width: 100%;
+  height: 38px;
+  padding: 7px 14px;
+  border-radius: 6px;
+  font-size: 15px;
```
Comment (lyudad) [inline src/components/Button/SignIn/styles.ts]: 1.3.2

File: src/components/Button/UniversalButton/UniversalButton.tsx [hunk 5]
```diff
@@ -1,95 +1,109 @@
...
+  ButtonVariant,
+  ButtonSize,
+} from '@/assets/styles/constants/buttonConstants';
+
+const paddingHorizontal = '16px';
+const borderRadius = '8px';
+const fontSizeDot = '20px';
+const opacityDisabled = 0.6;
+const opacityHover = 0.8;
+const defaultColor = '#fff';
```
Comment (lyudad) [inline src/components/Button/UniversalButton/UniversalButton.tsx]: not here, it should be in place where all the colors

File: src/components/Button/UniversalButton/styles.ts [hunk 6]
```diff
@@ -1,76 +1,94 @@
...
+import { FONTS } from '@/assets/styles/constants/fonts.ts';
+import { SIZES } from '@/assets/styles/constants/sizes.ts';
 
 type ButtonVariantType = keyof typeof buttonVariants;
 type ButtonColorType = keyof typeof buttonColors;
 
+const paddingHorizontal = '16px';
+const borderRadius = '8px';
+const outlineWidth = '3px';
+const opacityDisabled = 0.6;
```
Comment (lyudad) [inline src/components/Button/UniversalButton/styles.ts]: this variable is multiple files, why you doing this duplication, move it to some general file

File: src/components/Inputs/SignIn/styles.tsx [hunk 7]
```diff
@@ -1,22 +1,28 @@
...
+  && {
+    margin-bottom: 30px;
+  }
+
+  width: 100%;
+  height: 56px;
+
+  .MuiOutlinedInput-root {
+    &.Mui-focused fieldset {
+      border-color: #49454f;
```
Comment (lyudad) [inline src/components/Inputs/SignIn/styles.tsx]: 1.3.2

File: src/components/Inputs/SignIn/styles.tsx [hunk 8]
```diff
@@ -1,22 +1,28 @@
...
     }
-`;
\ No newline at end of file
+  }
+  .MuiInputLabel-root {
+    &.Mui-focused {
+      color: #625b71;
```
Comment (lyudad) [inline src/components/Inputs/SignIn/styles.tsx]: 1.3.2

File: src/components/LogoWithIcons/LogoTitle/styles.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+import { FONTS } from '@/assets/styles/constants/fonts';
+
+interface TitleProps {
+  color: string;
+}
+
+export const Title = styled.div<TitleProps>`
+  font-family: ${FONTS.AKSHAR};
+  font-size: 30px;
```
Comment (lyudad) [inline src/components/LogoWithIcons/LogoTitle/styles.ts]: 1.3.2

File: src/components/Titles/SignIn/styles.tsx
Comment (lyudad) [inline src/components/Titles/SignIn/styles.tsx:1]: I assume thatt "sign in" used only on sign in page, in this case it should be not in general "components" page, in should be in /pages/signIn/components

File: src/pages/SignInPage/styles.ts [hunk 10]
```diff
@@ -1,29 +1,46 @@
...
-  justify-content: center;
-  flex: 1;
+  flex-direction: column;
+  justify-content: space-evenly;
+  width: 653px;
+  min-height: 381px;
+  padding: 40px;
+  border-radius: 8px;
+  box-shadow: 0px 10px 30px rgba(0, 0, 0, 0.1);
+  background-color: #fff;
```
Comment (lyudad) [inline src/pages/SignInPage/styles.ts]: 1.3.2

File: src/pages/SignInPage/styles.ts [hunk 11]
```diff
@@ -1,29 +1,46 @@
...
-  align-items: center;
-  justify-content: center;
-  flex: 1;
+  flex-direction: column;
+  justify-content: space-evenly;
+  width: 653px;
+  min-height: 381px;
+  padding: 40px;
+  border-radius: 8px;
+  box-shadow: 0px 10px 30px rgba(0, 0, 0, 0.1);
```
Comment (lyudad) [inline src/pages/SignInPage/styles.ts]: 1.3.2

**Review by lyudad — APPROVED** [review]

**Pull request merged**